test/jdk/java/awt/dnd/RejectDragDropActionTest.java

```diff
@@ -97,14 +97,11 @@ public static void main(String[] args) throws Exception {
 
             robot.mouseMove(startPoint.x, startPoint.y);
             robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
-            for (Point p = new Point(startPoint); !p.equals(endPoint);
+            for (Point p = new Point(startPoint);
+                 !p.equals(endPoint) || incorrectActionDetected;
```

Comment:
```suggestion
                 !p.equals(endPoint) && !incorrectActionDetected;
```
You keep iterating while both conditions are true. As soon as either is false, exit the loop.